lib/fabrik/VirtualHostInstance.js

```diff
@@ -0,0 +1,93 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const BaseInstance = require('./BaseInstance');
+const bosh = require('../bosh');
+const utils = require('../utils');
+const mapper = require('./VirtualHostRelationMapper');
+const catalog = require('../models').catalog;
+
+class VirtualHostInstance extends BaseInstance {
+
+  constructor(guid, manager) {
+    super(guid, manager);
+    this.director = bosh.director;
+    this.mapper = mapper.VirtualHostRelationMapper;
+  }
+
+  get async() {
+    return false;
+  }
+
+  initialize() {
+    return this.getDeploymentNameForVirtualHostInstanceId(this.guid)
+      .then((deploymentName) => this.deploymentName = deploymentName);
+  }
+
+  create(params) {
+    const rabbitmqInstanceName = params.parameters.dedicated_rabbitmq_instance;
+    return this.cloudController.getMultiTenantServiceInstanceIdWithName(rabbitmqInstanceName, params.organization_guid, params.space_guid, 'rabbitmq')
+      .then((serviceInstanceId) => {
+        return this.director.getDeploymentNameForInstanceId(serviceInstanceId);
+      })
+      .then((deploymentName) => {
+        return this.manager
+          .createOrUpdateVirtualHost(deploymentName, params, this.guid);
+      });
+  }
+
+  delete(params) {
+    return this.initialize()
+      .then(() => this.manager.deleteVirtualHost(this.deploymentName, this.guid, params));
+  }
+
+  bind(params) {
+    return this
+      .initialize()
+      .then(() => this.manager.createBinding(this.deploymentName, this.guid, {
+        id: params.binding_id,
+        parameters: params.parameters || {}
+      }));
+  }
+
+  unbind(params) {
+    return this
+      .initialize()
+      .then(() => this.manager.deleteBinding(this.deploymentName, this.guid, params.binding_id));
+  }
+
+  getInfo() {
+    return this.initialize().then(() => {
+      this.parentInstanceId = this.getParentInstanceId();
+      return Promise
+        .all([
+          this.cloudController.getServiceInstance(this.guid),
+          this.cloudController.getServiceInstance(this.parentInstanceId),
+          this.cloudController.findServicePlanByInstanceId(this.parentInstanceId)
+          .then((body) => {
+            return new Promise.resolve(catalog.getPlan(body.entity.unique_id));
+          })
+        ])
+        .spread((instance, parent_instance, parent_instance_plan) => ({
+          title: `${this.plan.service.metadata.displayName || 'Service'} Dashboard`,
+          plan: this.plan,
+          service: this.plan.service,
+          instance: instance,
+          parent_plan: parent_instance_plan,
+          parent_instance: parent_instance
+        }));
+    });
+  }
+
+  getDeploymentNameForVirtualHostInstanceId(id) {
+    return this.mapper.getDeploymentName(`virtual_host_instance-${id}`);
```

Comment:
`virtual_host_instance-${` seems unnecessary. The method `getDeploymentName` signature second parameter says `isntance_guid`, however u seem to have prefixed it with some constant. This can be removed and can pass the instance id as is.